extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/query/SharedVisibilitySplittingIterator.java

```diff
@@ -0,0 +1,102 @@
+package mil.nga.giat.geowave.datastore.accumulo.query;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import mil.nga.giat.geowave.datastore.accumulo.util.AccumuloUtils;
+
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.PartialKey;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.iterators.user.TransformingIterator;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.io.Text;
+
+public class SharedVisibilitySplittingIterator extends
+		TransformingIterator
+{
+	// must execute prior to WholeRowIterator/QueryFilterIterator
+	public static final int ITERATOR_PRIORITY = QueryFilterIterator.WHOLE_ROW_ITERATOR_PRIORITY - 2;
+	public static final String ITERATOR_NAME = "SHRD_VIZ_SPLT_ITERATOR";
+
+	@Override
+	protected PartialKey getKeyPrefix() {
+		// this iterator may modify CQ, but will not modify CF
+		return PartialKey.ROW_COLFAM;
+	}
+
+	@Override
+	protected void transformRange(
+			final SortedKeyValueIterator<Key, Value> input,
+			final KVBuffer output )
+			throws IOException {
+
+		while (input.hasTop()) {
+
+			final Key currKey = input.getTopKey();
+			final Value currVal = input.getTopValue();
+
+			// check to see if the current value is a composite field containing
+			// multiple attributes with shared visibility
+			if (Arrays.equals(
```

Comment:
Agreed.  Said the exact same thing to Steve on Friday.  Will be curious to see what impact it has.